src/python/WMCore/Database/ExecuteDAO.py

```diff
@@ -160,16 +160,22 @@ def __init__(self, logger=None, configFile=None,
             config.CoreDatabase.dialect = getBackendFromDbURL(os.getenv("DATABASE"))
             config.CoreDatabase.socket = os.getenv("DBSOCK", None)
 
+        # always try to determine the dilect from the URL
+        if getattr(config.CoreDatabase, "connectUrl", None):
+            config.CoreDatabase.dialect = getBackendFromDbURL(config.CoreDatabase.connectUrl)
```

Comment:
Given this code, I would suggest to drop the "dialect" parameter. If user provides the `connectUrl`, then the dialect can be infered from it. It will make things simpler.

Reply:
done